cpp/src/arrow/dataset/file_csv.h

```diff
@@ -83,6 +82,35 @@ struct ARROW_DS_EXPORT CsvFragmentScanOptions : public FragmentScanOptions {
   csv::ReadOptions read_options = csv::ReadOptions::Defaults();
 };
 
+class ARROW_DS_EXPORT CsvFileWriteOptions : public FileWriteOptions {
+ public:
+  /// Options passed to csv::MakeCSVWriter. use_threads is ignored
```

Comment:
Is use_threads used elsewhere?  The way the code is structured threads could be used for the casts, so if it is important we might want to file a follow-up JIRA.

Reply:
I copied this from the equivalent IPC struct - it doesn't apply here since there's no such parameter of course.